### PR TITLE
docs(operate): document multi-tenancy

### LINKED ISSUE #2702: Operate: document multi-tenancy for self-managed
### Description

We need to document new multi-tenancy feature and how to configure it in Operate.

* configuration parameter to switch the feature on and off
* consequences of disabled flag (all users can only see `<default>` tenant data)
* consequences of enabled flag (all users can only see tenant data as authorized by identity)
* necessity to configure Zeebe authentication env vars
* migration from previous versions (all data will be assigned to `<default>` tenant

### PR DESCRIPTION
**Pull request by sdorokhova**
closes #2702

## Description

- configuration parameter to switch the feature on and off
- consequences of disabled flag (all users can only see <default> tenant data)
- consequences of enabled flag (all users can only see tenant data as authorized by identity)
- necessity to configure Zeebe authentication env vars
- long migration warning

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
@ralfpuchert are you able to review? Sounds like we'd like to merge this quickly and release.

**Review by mihail-ca — DISMISSED** [review]

File: docs/apis-tools/operate-api/overview.md [hunk 1]
```diff
@@ -126,6 +126,19 @@ curl -b cookie.txt -X POST 'http://localhost:8080/v1/process-definitions/search'
...
+:::note
+[Multi-tenancy feature](../../self-managed/operate-deployment/operate-configuration.md#multi-tenancy) is only available in Self-managed setup.
+:::
+
+All Operate endpoints for which tenant assignment is relevant will:
+
+- return `tenantId` field in response
+- provide `tenantId` search parameter
+
+Check Swagger docs for exact request and response structure.
```
Comment (christinaausley) [inline docs/apis-tools/operate-api/overview.md:140]: We're going to need to link to Swagger documentation here.

File: docs/self-managed/operate-deployment/operate-configuration.md [hunk 2]
```diff
@@ -53,6 +54,35 @@ To change the values for http header for security reasons, you can use the confi
...
+the feature can be found in [general Multi-tenancy description](../concepts/multi-tenancy.md).
+
+The following configuration is required to enable multi-tenancy in Operate:
+
+| Name                                 | Description                                                  | Default value |
+| ------------------------------------ | ------------------------------------------------------------ | ------------- |
+| camunda.operate.multiTenancy.enabled | Activates the multi-tenancy feature within the Tasklist app. | false         |
+
+:::caution
+To ensure seamless integration and functionality, the multi-tenancy feature should also be enabled across all associated components. This is done using their specific multi-tenancy feature flags.
```
Comment (christinaausley) [inline docs/self-managed/operate-deployment/operate-configuration.md:69]: Can we link to more information on this?

File: docs/self-managed/operational-guides/update-guide/820-to-830.md [hunk 3]
```diff
@@ -119,6 +119,11 @@ be shortened, set the following properties to `false` while migrating Zeebe to `
   - Broker: `ZEEBE_BROKER_EXPERIMENTAL_CONSISTENCYCHECKS_ENABLEPRECONDITIONS`
   - Broker: `ZEEBE_BROKER_EXPERIMENTAL_CONSISTENCYCHECKS_ENABLEFOREIGNKEYCHECKS`
 
+### Operate migration
+
+Updating may take longer as almost all the Operate data must be reindexed in Elasticsearch to add the `tenantId` field. To speed up data migration during version update
+consider adding more resources to Elasticsearch cluster.
```
Comment (christinaausley) [inline docs/self-managed/operational-guides/update-guide/820-to-830.md:125]: What do you mean by "more resources?"

**Review by christinaausley — CHANGES_REQUESTED** [review]
A few comments for clarification. This should also be added to `8.3` docs, yes?

**Comment by akeller** [comment]
> A few comments for clarification. This should also be added to `8.3` docs, yes?

Yes, this is part of 8.3 functionality.

**Comment by christinaausley** [comment]
For purposes of timeliness I'm going to backport and merge, but @sdorokhova we can adjust any comments as necessary in a separate PR.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**